anonymizer.go

```diff
@@ -0,0 +1,44 @@
+package goanonymizer
+
+import (
+	"errors"
+	"net"
+)
+
+const DefaultIPv4Mask = "255.255.255.0"
+const DefaultIPv6Mask = ""ffff:ffff:ffff:ffff:0000:0000:0000:0000""
+//Anonymize struct that hold mask for V4 and V6.
+type Anonymize struct {
+	v4Mask net.IPMask
+	v6Mask net.IPMask
+}
+
+//NewAnonymize returns instance of Anonymize.
+//If v4Mask and v6Mask are not provided, default values are used.
+func NewAnonymize(v4Mask, v6Mask string) *Anonymize {
+	ipV4 := net.ParseIP(v4Mask)
+	if ipV4 == nil {
+		ipV4 = net.ParseIP("255.255.255.0")
+	}
+	
+	ipV6 := net.ParseIP(v6Mask)
+	if ipV6 == nil {
+		ipV6 = net.ParseIP("ffff:ffff:ffff:ffff:0000:0000:0000:0000")
+	}
+
+	return &Anonymize{v4Mask: net.IPMask(ipV4.To4()), v6Mask: net.IPMask(ipV6)}
+}
+
+//AnonymizeIp gets IP and transform it based on v4Mask and v6Mask.
+func (a Anonymize) AnonymizeIp(ipStr string) (string, error) {
```

Comment:
Using the same method for IPv4 and IPv6 addresses does not work as expected.
If an Ipv6 address is in the IPv4-mapped IPv6 address range, `To4()` will convert the Ipv6 address to an IPv4 address.

It also does not work to first call `To16()` and only `To4()` if To16 returns nil,
`To16()` converts IPv4 addresses to their IPv6 representation.

I'll add a method per IP version instead